packages/track-event-mixin/track-event-gesture.ts

```diff
@@ -0,0 +1,308 @@
+// Check for touch-only devices
+const IS_TOUCH_ONLY = navigator.userAgent.match(/iP(?:[oa]d|hone)|Android/);
+// Number of last N track positions to keep
+const TRACK_LENGTH = 2;
+// Radius for track
+const TRACK_DISTANCE = 5;
+// Stored touch identifier
+const TOUCH_STATE = {
+  id: -1
+};
+
+type TrackMove = {
+  x: number;
+  y: number;
+};
+
+type TrackGestureInfo = {
+  x: number;
+  y: number;
+  state: 'start' | 'track' | 'pan-x' | 'pan-y' | 'end';
+  started: boolean;
+  moves: TrackMove[];
+  addMove: (move: TrackMove) => void;
+  moveFn?: (event: MouseEvent) => void;
+  upFn?: (event: MouseEvent) => void;
+};
+
+type TrackGesture = {
+  name: string;
+  touchAction: string;
+  deps: Array<'mousedown' | 'touchstart' | 'touchmove' | 'touchend'>;
+  info: TrackGestureInfo;
+  reset: () => void;
+  mousedown: (event: MouseEvent) => void;
+  touchstart: (event: TouchEvent) => void;
+  touchmove: (event: TouchEvent) => void;
+  touchend: (event: TouchEvent) => void;
+};
+
+function isMouseEvent(name: string): boolean {
+  return ['mousedown', 'mousemove', 'mouseup'].indexOf(name) > -1;
+}
+
+function hasLeftMouseButton(ev: MouseEvent): boolean {
+  const { type } = ev;
+  let result;
+  if (type === 'mousemove') {
+    // ev.button is not reliable for mousemove (0 is overloaded as both left button and no buttons)
+    // instead we use ev.buttons (bitmask of buttons): 0 for no buttons, 1 for left button
+    // NOTE: allow undefined for testing events
+    const buttons = ev.buttons === undefined ? 1 : ev.buttons;
+    // buttons is a bitmask, check that the left button bit is set (1)
+    result = Boolean(buttons & 1); // eslint-disable-line no-bitwise
+  } else {
+    // NOTE: allow undefined for testing events
+    const button = ev.button === undefined ? 0 : ev.button;
+    // ev.button is 0 in mousedown/mouseup/click for left button activation
+    result = button === 0;
+  }
+  return result;
+}
+
+function trackHasMovedEnough(info: TrackGestureInfo, x: number, y: number): boolean {
+  if (info.started) {
+    return true;
+  }
+  const dx = Math.abs(info.x - x);
+  const dy = Math.abs(info.y - y);
+  return dx >= TRACK_DISTANCE || dy >= TRACK_DISTANCE;
+}
+
+function trackFire(info: TrackGestureInfo, target: EventTarget, touch: Touch | MouseEvent) {
```

Comment:
I think `fireTrack` would sound more natural for this. Not a blocker.

Reply:
Done.